manager.go

```diff
@@ -58,12 +61,18 @@ func (sm *spinnerManager) AddSpinner(message string) *Spinner {
 		HasUpdate:     sm.hasUpdate,
 	}
 	spinner := NewSpinner(opts)
+
+	sm.spinnersMu.Lock()
 	sm.spinners = append(sm.spinners, spinner)
+	sm.spinnersMu.Unlock()
```

Comment:
Could we `defer` this above the call to append?

Reply:
Done!